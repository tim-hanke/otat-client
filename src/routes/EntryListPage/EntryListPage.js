import React, { Component } from "react";
import ArticleListContext from "../../contexts/ArticleListContext";
import EntryApiService from "../../services/entry-api-service";
import { Section } from "../../components/Utils/Utils";
import EntryListItem from "../../components/EntryListItem/EntryListItem";
import "./EntryListPage.css";

export default class EntryListPage extends Component {
  static contextType = ArticleListContext;

  componentDidMount() {
    this.context.clearError();
    EntryApiService.getEntries()
      .then(this.context.setEntryList)
      .catch(this.context.setError);
  }

  renderArticles() {
    const { entryList = [] } = this.context;
    return entryList
      .sort((a, b) => (a.id < b.id ? 1 : -1))
      .map((article) => <EntryListItem key={article.id} article={article} />);
  }

  render() {
    const { error } = this.context;
    return (
      <Section list className="EntryListPage">
        {error ? (
          <p className="red">There was an error, try again</p>
        ) : (
          this.renderArticles()
        )}
      </Section>
    );
  }
}
