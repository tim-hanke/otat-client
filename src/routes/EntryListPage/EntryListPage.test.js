import React from "react";
import ReactDOM from "react-dom";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import EntryListPage from "./EntryListPage";

describe("EntryListPage component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <EntryListPage />
      </MemoryRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <EntryListPage />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
