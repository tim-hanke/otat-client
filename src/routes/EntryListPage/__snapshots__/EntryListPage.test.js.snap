// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryListPage component renders the UI as expected 1`] = `
<section
  className="Section Section--list EntryListPage"
/>
`;
